### PR TITLE
Font Library: refactor FontsGrid component

### PR DESCRIPTION
**Pull request by t-hamano**
## What?

This PR refactors the FontsGrid component to improve code quality.

## Why?

The current `FontsGrid` component receives `children`. In order to pass the `children` to the `FontsGrid` component, we render the `FontCard` component using the `map()` function. Inside the `FontsGrid `component, we loop through its children again, causing a double loop.

We should simply pass the fonts as props to the `FontsGrid` component, and the `FontsGrid` component should render based on those props.

## How?

At the same time as refactoring, I integrated `LibraryFontCard` and `FontCard` components. This is because the component names are similar and difficult to identify. Also, the `FontCard` component is completely dependent on the `LibraryFontCard` component, so there shouldn't be any need to split it.

## Testing Instructions

The layout and behavior of the Font Library should remain unchanged.

### CODE REVIEW

File: packages/edit-site/src/components/global-styles/font-library-modal/fonts-grid.js [hunk 1]
```diff
@@ -8,11 +8,17 @@ import {
 } from '@wordpress/components';
 import { useState, useEffect } from '@wordpress/element';
 
-function FontsGrid( { title, children, pageSize = 32 } ) {
+/**
+ * Internal dependencies
+ */
+import FontCard from './font-card';
+
+const PAGE_SIZE = 32;
```
Comment (t-hamano) [inline packages/edit-site/src/components/global-styles/font-library-modal/fonts-grid.js:16]: The `pageSize` prop that the FontsGrid component has is never used, so I defined it as a constant.

File: packages/edit-site/src/components/global-styles/font-library-modal/fonts-grid.js [hunk 2]
```diff
@@ -40,16 +46,16 @@ function FontsGrid( { title, children, pageSize = 32 } ) {
...
+					{ fonts.slice( 0, itemsLimit ).map( ( font, i ) => (
+						<FontCard
+							key={ i }
+							font={ font }
+							ref={
+								i === itemsLimit - 1 ? setLastItem : undefined
+							}
+							onChange={ onChange }
+						/>
+					) ) }
```
Comment (t-hamano) [inline packages/edit-site/src/components/global-styles/font-library-modal/fonts-grid.js:58]: This refactoring allows us to remove unnecessary `div` elements.

**Review by matiasbenedetto — COMMENTED** [review]
Hi @t-hamano, thanks for submitting this PR :)

I've tested it, and it's working fine, but I'm not sure if it's still relevant given the on-going [implementation of the pagination](https://github.com/WordPress/gutenberg/pull/58794) as requested in [this issue](https://github.com/WordPress/gutenberg/issues/58067).

@pbking, could you take a look at this and confirm, please?

**Comment by t-hamano** [comment]
@matiasbenedetto I think #58794 will definitely have a higher priority. Therefore, I'm fine with #58794 being advanced before this PR. Once #58794 is merged, I would like to resolve conflicts 👍

**Comment by pbking** [comment]
Indeed, I believe this won't be relevant if we land the pagination branch; that does away with the FontsGrid component.  There might be other improvements in there that are relevant and useful.

**Comment by t-hamano** [comment]
This PR now has too many conflicts with trunk. Therefore, I would like to close this PR and submit a new PR if necessary.

**Pull request closed without merging**